nsot/models.py

```diff
@@ -1470,6 +1494,131 @@ def to_dict(self):
         }
 
 
+class Circuit(Resource):
+    """Represents two network Interfaces that are connected"""
+
+    # a_endpoint interface
+    a_endpoint = models.OneToOneField(
+        Interface,
+        on_delete=models.CASCADE,
+        db_index=True,
+        null=False,
+        related_name='circuit_a',
+        verbose_name='A-side Interface',
+        help_text='Unique ID of interface at the A-side'
+    )
+
+    # z_endpoint interface
+    z_endpoint = models.OneToOneField(
+        Interface,
+        on_delete=models.CASCADE,
+        db_index=True,
+        null=True,
+        related_name='circuit_z',
+        verbose_name='Z-side Interface',
+        help_text='Unique ID of interface at the Z-side'
+    )
+
+    # We are currently inferring the site_id from the parent (A-side) Interface
+    # in the .save() method
+    site = models.ForeignKey(
+        Site, db_index=True, related_name='circuits',
+        on_delete=models.PROTECT,
+        help_text='Unique ID of the Site this Circuit is under'
+    )
+
+    name = models.CharField(
+        max_length=255, unique=True,
+        help_text='Unique display name of circuit',
+        default='',
+    )
+
+    def __unicode__(self):
+        return u'%s' % self.name
+
+    @property
+    def interfaces(self):
+        intf_list = [self.a_endpoint, self.z_endpoint]
+        intf_list = [i for i in intf_list if i]
+        return intf_list
+
+    @property
+    def addresses(self):
+        interfaces = self.interfaces
```

Comment:
Yeah, maybe the Django ORM can support that lookup. Could be done with SQL pretty easily.